src/index.js

```diff
@@ -171,6 +195,15 @@ function parseFirmwareImage(firmware) {
  * @returns {Promise} resolved to prepared device
  */
 async function prepareInDFUBootloader(device, dfu) {
+
+    if (!isDeviceInDFUBootloader(device)) {
+        const usbdev = device.usb.device;
+        debug('Switching to bootloader mode.');
+        const newSerNr = await predictSerialNumberAfterReset(usbdev);
+        debug('Serial number after reset should be:', newSerNr);
+        await detachAndWaitFor(usbdev, getDFUInterfaceNumber(usbdev), newSerNr);
```

Comment:
This breaks the logic, there will be a new device with a new serialport and following lines will fail to find the old one.